infrastructure/src/main/java/org/corfudb/infrastructure/CorfuMsgHandler.java

```diff
@@ -209,43 +211,36 @@ private void registerMethod(@Nonnull final MethodHandles.Lookup caller,
     private Handler<CorfuMsg> generateConditionalHandler(@Nonnull final AbstractServer server,
             @Nonnull final CorfuMsgType type,
             @Nonnull final Handler<CorfuMsg> handler) {
-        if (!MetricsUtils.isMetricsCollectionEnabled()) {
-            // If metrics are disabled, register the handler without instrumentation.
-            return (msg, ctx, r) -> {
-                if (server.isShutdown()) {
-                    log.warn("Server received {} but is shutdown.", msg.getMsgType().toString());
-                    r.sendResponse(ctx, msg, CorfuMsgType.ERROR_SHUTDOWN_EXCEPTION.msg());
-                    return;
-                }
-
-                if (!server.isServerReadyToHandleMsg(msg)) {
-                    r.sendResponse(ctx, msg, CorfuMsgType.NOT_READY.msg());
-                    return;
-                }
+        // Generate a timer based on the Corfu message type
+        final Timer timer = getTimer(type);
```

Comment:
Is this timer shared between messages of the same time, or is a new timer created every time?

Reply:
Timers get registered to metrics registry and reused.